### ISSUE TITLE
Mochad X10 Binding: Can't get CM19A (RF) to work

### ISSUE THREAD

**Issue by dannyboyw** [openhab/openhab#3536]
I have a cron job that has consistently used mochad for several years, therefore I have ruled out issues with it.  I am doing the following while trying to debug my problem: 
1. Telnet into mochad, so I can see all that is being sent/received.
2. Telnet into openhab osgi console to cutout the UI (i.e. don't have to consider UI issues, but I still have a sitemap config below to do final test if I can get it working).  I send commands to mochad via the console like this: `osgi> openhab send Appl_Module_Test ON`
   ...and get: 
   `2015-12-11 07:10:24.810 [INFO ] [runtime.busevents             ] - Appl_Module_Test received command ON
   Command has been sent successfully.`
3. A very slim openhab config to hopefully simplify troubleshooting this issue.  I left NTP in to prove that other bindings are working.

default.items has:

```
Switch Appl_Module_Test         "Appliance Test Module"         {mochadx10="a16:rf"}
Switch Transceiver_Hse_Cd_A     "House Code A Transceiver"      {mochadx10="a1:rf"}

DateTime    Date    "Date and Time: [%1$tA, %1$td.%1$tm.%1$tY %1$tT]" <calendar>  {ntp=""}
```

default.sitemap looks like:

```
sitemap default label="Debug My X-10 House"
{

    Frame label="Date" {
      Text item=Date
      }

    Frame label="X-10 Controls" {
      Switch item=Appl_Module_Test label="Appliance Test Module"
      Switch item=Transceiver_Hse_Cd_A label="House Code A Transceiver Module"
      }
}
```

openhab.cfg has:

```
...
###################################### Mochad X10 ######################################
# IP address and port of the Mochad X10 server
mochadx10:hostIp=localhost
mochadx10:hostPort=1099
...
```

I do get a warning in the log about mochad:

```
...
2015-12-11 07:49:58.048 [DEBUG] [.o.b.ntp.internal.NtpActivator] - NTP binding has been started.
2015-12-11 07:49:58.080 [DEBUG] [i.internal.GenericItemProvider] - Start processing binding configuration of Item 'Date (Type=DateTimeItem, State=Uninitialized)' with 'NtpGenericBindingProvider' reader.
2015-12-11 07:49:58.129 [INFO ] [.service.AbstractActiveService] - NTP Refresh Service has been started
2015-12-11 07:49:58.135 [DEBUG] [.m.internal.MochadX10Activator] - Mochad X10 binding has been started.
2015-12-11 07:49:58.181 [DEBUG] [i.internal.GenericItemProvider] - Start processing binding configuration of Item 'Appl_Module_Test (Type=SwitchItem, State=Uninitialized)' with 'MochadX10GenericBindingProvider' reader.
2015-12-11 07:49:58.241 [DEBUG] [b.n.i.NetworkUpsToolsActivator] - NetworkUpsTools binding has been started.
2015-12-11 07:49:58.309 [WARN ] [i.internal.GenericItemProvider] - Attempted to register a second BindingConfigReader of type 'mochadx10'. The primaraly reader will remain active!
...
```

I installed openhab and bindings using apt on my Ubuntu box.  I have been searching for a robust HA platform with a highly adaptable UI to control my old X-10 stuff for a long time, but that will also let me use newer technologies.  I think openhab might be the one, if I can just get this working.

Thank you very much for your help!
##

**Comment by robertesorensen** [comment]
Did you ever figure this out? I am having the same issue?

**Comment by dannyboyw** [comment]
No, but I have not dug any further, except for this request and searching the interwebs.  I thought I meticulously followed all the documentation I could find, so I am hoping it is some little (undocumented) detail specific to the CM19A that I did not do.  I remember reading a couple threads w/ other saying they got CM19A working with this binding, so I was hoping they would chime in.

**Comment by dannyboyw** [comment]
I added in [community forum](https://community.openhab.org/t/need-help-with-mochad/5887/7) in hopes that going full circle will help find a resolution.  It appears that I am not the only one experiencing issues

**Comment by federic0** [comment]
I am running OH 1.8 (and 1.7 before) with a CM15A and do not have this problem. I do not have mochad/CM15A on the same host as OH, have you tried changing that to an external host see if that changes anything?

**Comment by dannyboyw** [comment]
Thank you for your insight!  I am running OH 1.8 now as well.  No dice on changing mochad to remote host.  I did independent testing at all points (mochad working on remote host, OH host sends/receives CLI commands to/from mochad remote host, etc...) prior to testing in OH.  

Can rfxcom receive X10 commands?  I also have an old [W800RF32A](http://www.wgldesigns.com/w800.html) and [FireCracker CM17A](http://www.thehomeautomationstore.com/cm17a.html).  See any potential for a stable send/receive X10 setup with any combo of these devices w/o too much time consuming hackery?

Thanks again!

**Comment by federic0** [comment]
Hi,
rfxcom will receive X10 commands as long as they are radio (rf) commands and no power line (pl). I have no knowledge on the other equipment you have.
Cheers

**Comment by dannyboyw** [comment]
It appears that I would have to buy ~$100 piece of hardware to make rfxcom
work.  I would prefer to get something that works w/ what I have.  I want
to eventually phase out the X10 stuff.  Rfxcom does look versatile, in that
it has the capabilities to work with many non-X10 RF modules (wx, motion,
etc), but not sure I want to lay down $100 for it.  I am still trying to
figure out my next move, as far as Insteon vs z-wave.  I was thinking
potentially both, which is the beauty of OH.  If I got an insteon device
that works w/ OH, but is also bward comptl. w/ X10 this headache might go
away.  I do have major concerns that cm19a purportatly works with this
binding and I can't seem to get help with making it work.

**Comment by federic0** [comment]
rfxcom is expensive but you get the possibility to interact with all these sensors and actuators that are quite cheap (Coco - now Trust, many thermometers including many Oregon Scientific, etc.).
If you are thinking Insteon, the Insteon gateway should be able to send X10 commands. I have no experience with Insteon and OH though.

**Comment by Crenel** [comment]
I would not recommend Insteon. I have struggled to get any value from my investment in their hardware and the X10 "support" has been a dead end. A couple times I was able to toggle an X10 module via Insteon but a) it took a lot of hunting for a power outlet that allowed any connection at all and b) it was inconsistent, only working maybe 60% of the time. Also, the newer Insteon hub apparently doesn't support X10 at all, so you'd want the older one, but they're no longer updating the firmware for the older one and just push people to the newer one. Also, I have not yet been able to connect OH to my Insteon hub, although I'm new enough to OH that this could be something I'm missing or doing wrong. Anyway, for me, Insteon = buyer's remorse.
